client.go

```diff
@@ -123,8 +129,12 @@ func (client *Client) KeyID() string {
 	return client.keyID
 }
 
-func (client *Client) init(ctx context.Context) {
-	client._init.Do(func() {
+func (client *Client) cache(ctx context.Context) {
+	client.init.Do(func() {
+		var err error
+		ctx, span := client.startSpan(ctx, "cache")
+		defer func() { span.End(err) }()
+
 		response, err := client.remoteClient.GetKey(ctx, client.keyName, client.keyVersion, nil)
```

Comment:
Note that this will include a child span of the SDK call as we don't have suppression implemented yet (see the [guidelines](https://azure.github.io/azure-sdk/general_implementation.html#distributed-tracing)).

Reply:
That's actually what I want - that this tracing is inclusive of any the azkeys module traces. Or did I misunderstand you?

Reply:
Let me clarify.  In this case, while the span for `GetKey` will be omitted, any inner spans it would create (e.g. the HTTP span) will show up as a child span of `cache`.

I have a fix to suppress the nested API call spans which describes this better [here](https://github.com/Azure/azure-sdk-for-go/pull/21251).

Reply:
I would actually want the `GetKey` span, though. Would that be solved by me using `runtime.StartSpan` as you suggested above?

Reply:
If you want the `GetKey` span, then you shouldn't use `runtime.StartSpan` as it will automatically suppress it (once my fix goes in).

Reply:
But if you're wanting to follow the SDK design guidelines then you should suppress it.

Reply:
That seems odd. Why would I want to suppress the important bits? The HTTP request and, more importantly, the response are always good to trace.

Reply:
An API call creates two spans; a span with the API name (e.g. "GetKey") and a child span with the HTTP request/response.  The only span that gets suppressed is the "outer" span with the API name.  So, in your trace, you'd see a span named `cache` with a child span that contains the HTTP request/response, not three spans e.g. cache->GetKey->HTTP request/response.

Reply:
Doesn't that presume that an intermediate API call like `GetKey` does nothing else of value? In .NET, we trace the entire call chain - even simple forward-only helpers.

Reply:
That seems to be in contradiction to the [design guidelines](https://azure.github.io/azure-sdk/general_implementation.html#distributed-tracing).
[✅](https://azure.github.io/azure-sdk/general_implementation.html#general-tracing-suppress-client-spans-for-inner-methods) DO When client method creates a new span and internally calls into other public client methods of the same or different Azure SDK, spans created for inner client methods MUST be suppressed, their attributes and events ignored. Nested spans created for REST calls MUST be the children of the outer client call span. Suppression is generally done by Azure Core.

Reply:
@annelo-msft or @joshlove-msft do I remember wrong, or do we not suppress intermediate API calls between convenience methods and actual REST calls? I recall, at least in Key Vault, seeing them both.